src/wallet/rpcdump.cpp

```diff
@@ -114,6 +112,9 @@ UniValue importprivkey(const UniValue& params, bool fHelp)
     if (params.size() > 2)
         fRescan = params[2].get_bool();
 
+    if (fRescan && fPruneMode)
+        throw JSONRPCError(RPC_WALLET_ERROR, "Rescan is disabled in pruned mode");
```

Comment:
You could just say "-rescan is disabled in pruned mode" which would be more clear (to me at least :)